atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -295,12 +302,13 @@ private void checkNamespaceConfig() {
                     Preconditions.checkState(client.equals(namespace),
                             "If present, the TimeLock client config should be the same as the"
                                     + " atlas root-level namespace config.")));
+            return namespace;
         } else if (!(keyValueService() instanceof InMemoryAtlasDbConfig)) {
             Preconditions.checkState(keyValueService().namespace().isPresent(),
                     "Either the atlas root-level namespace"
                             + " or the keyspace/dbName/sid config needs to be set.");
 
-            Optional<String> keyValueServiceNamespace = keyValueService().namespace();
+            String keyValueServiceNamespace = keyValueService().namespace().get();
```

Comment:
You can minimise the diff by not extracting this until you return it.  But I don't mind

Reply:
But then we are doing the get multiple times :(